packages/tldraw/src/lib/shapes/arrow/ArrowShapeUtil.tsx

```diff
@@ -790,6 +792,56 @@ export class ArrowShapeUtil extends ShapeUtil<TLArrowShape> {
 			},
 		]
 	}
+
+	override getInterpolatedProps(
+		startShape: TLArrowShape,
+		endShape: TLArrowShape,
+		progress: number
+	): TLArrowShapeProps {
+		const discretePropKeys = [
+			'labelColor',
+			'color',
+			'fill',
+			'dash',
+			'size',
+			'arrowheadStart',
+			'arrowheadEnd',
+			'font',
+			'text',
+		]
+
+		const isOverHalfway = progress > 0.5
+		const sourceShape = isOverHalfway ? endShape : startShape
+
+		const discreteProps: Partial<TLArrowShapeProps> = {}
+
+		discretePropKeys.forEach((propKey) => {
+			;(discreteProps as any)[propKey] = (sourceShape.props as any)[propKey]
```

Comment:
Not sure about this, but neither me nor @MitjaBezensek could come up with anything better.

Reply:
the issue with this is if we add/remove props to the arrow shape, this won't give us errors. what about:

```tsx
function interpolateDiscreate<R extends { props: object }, P extends keyof T['props']>(start: R, end: R, prop: P, progress: number) {
    if (progress < 0.5) return start.props[prop]
    return end.props[prop]
}
```

and then listing out the things you want to interpolate this way:
```tsx
return {
    props: {
        color: interpolateDiscrete(start, end, 'color', p),
        // etc
    }
}
```

Reply:
I proposed 
```typescript
function copyKeys(sourceShape: TLArrowShape) {
	const discreteProps: Partial<TLArrowShapeProps> = {}
	const keysToCopy = ['labelColor' as const, 'color' as const]

	keysToCopy.forEach((key) => {
		discreteProps[key] = sourceShape.props[key]
	})
	return discreteProps
}
```